text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+# Resource and Entities - Data Model Part 2
+
+This is a proposal to address Resource and Entity data model interactions,
+including a path forward to address immediate friction and issues in the
+current resource specification.
+
+<!-- toc -->
+
+<!-- tocstop -->
+
+## Motivation
+
+This proposal attempts to focus on the following problems within OpenTelemetry to unblock multiple working groups:
+
+- Allowing mutating attributes to participate in Resource ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).
+- Allow Resource to handle entities whose lifetimes don't match the SDK's lifetime ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).
+- Provide support for async resource lookup ([spec#952](https://github.com/open-telemetry/opentelemetry-specification/issues/952)).
+- Fix current Resource merge rules in the specification, which most implementations violate ([oteps#208](https://github.com/open-telemetry/oteps/pull/208), [spec#3382](https://github.com/open-telemetry/opentelemetry-specification/issues/3382), [spec#3710](https://github.com/open-telemetry/opentelemetry-specification/issues/3710)).
+- Allow semantic convention resource modeling to progress ([spec#605](https://github.com/open-telemetry/opentelemetry-specification/issues/605), [spec#559](https://github.com/open-telemetry/opentelemetry-specification/issues/559), etc).
+
+## Design
+
+### Approach - Resource Improvements
+
+Let's focus on outlining Entity detectors and Resource composition.  This has a higher priority for fixing within OpenTelemetry, and needs to be unblocked sooner. Then infer our way back to data model and Collector use cases.
+
+We define the following SDK components:
+
+- **Resource Detectors (legacy)**:  We preserve existing resource detectors.  They have the same behavior and interfaces as today.
+- **Entity Detectors (new)**: Detecting an entity that is relevant to the current instance of the SDK.  For example, this would detect a service entity for the current SDK, or its process. Every entity must have some relation to the current SDK.
+- **Resource Coordinator (new)**: A component responsible for taking Resource and Entity detectors and doing the following:
+  - Constructing a Resource for the SDK from detectors.
+  - Dealing with conflicts between detectors.
+  - Providing SDK-internal access to detected Resources for reporting via Log signal on configured LogProviders.
+  - *(new) Managing Entity changes during SDK lifetime, specifically dealing with entities that have lifetimes shorter than the SDK*
+
+#### Resource Coordinator
+
+The SDK Resource Coordinator is responsible for running all configured Resource and Entity Detectors.  There will be some (user-controlled, otel default) priority order to these.
```

Comment:
It sounds we need to document the set of known entity types, their associated attributes, and their relative priority. This seems like something for the `semantic-conventions` repo eventually. Is that what you had in mind?

Reply:
>It sounds we need to document the set of known entity types, their associated attributes ... This seems like something for the semantic-conventions repo eventually. Is that what you had in mind?

Yes, that's the plan.

> ... and their relative priority. 

That's a good question. I was thinking it will be an SDK configuration chosen by the end user, but perhaps it has a place in semconv?

Reply:
We MUST expose it to end users, but I think I agree, for OTEL itself we should have a default for these things.

I'd like to expand this as an open question for how to solve, but semconv is a likely candidate for that.

Reply:
If you want to understand what semconv already have - I just migrated `type` into the YAML here: https://github.com/open-telemetry/semantic-conventions/pull/1400

It's a good survey of what our community was building prior to this OTEP.